### PR TITLE
Uppercase vowels

### PR DESCRIPTION
**Pull request by xettrisomeman**

### CODE REVIEW

File: uppercase_vowels/exercise.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Level (beginner, intermediate, advanced): beginner
+Tags (2-5): string
+Dependencies (list any required libraries or leave as "None"):
+GitHub user to credit: xettrisomeman
+---
+
+# Uppercase Vowels
+
+---
+ The Python function `uppercase_vowels` takes in `string` or `filename` and returns `string` with vowels uppercased. The `string` can contain punctuation, uppercase consonants but in the final result vowels should be uppercased and consonants should be lowercased and punctuations should be removed while preserving spaces.
```
Comment (bbelderbos) [inline uppercase_vowels/exercise.md]: nice

Comment (bbelderbos) [reply uppercase_vowels/exercise.md]: we do have two vowel exercises but they are different:

image

File: uppercase_vowels/exercise.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+> Extension of `file` should not be uppercased.
+
+Examples of `extensions`:
```
Comment (bbelderbos) [inline uppercase_vowels/exercise.md:13]: nice twist

File: uppercase_vowels/script.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+def uppercase_vowels(text: str) -> str:
```
Comment (bbelderbos) [inline uppercase_vowels/script.py]: please move this file to script-template.py and add a new script.py with your solution code

File: uppercase_vowels/test_script.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from script import uppercase_vowels
+
+
+def test_uppercase_vowels_simple():
+    assert uppercase_vowels("abCDefghijkl") == "AbcdEfghIjkl"
+    assert uppercase_vowels("This test should pass itself") == "ThIs tEst shOUld pAss ItsElf"
+    assert uppercase_vowels("data.jpg foul") == "dAtA jpg fOUl"
+
+def test_uppercase_vowels_punctuation():
```
Comment (bbelderbos) [inline uppercase_vowels/test_script.py:13]: did you run pre-commit? I'd expect ruff to add 2 newlines between functions

**Review by bbelderbos — CHANGES_REQUESTED** [review]
Great start, please add the solution...

File: uppercase_vowels/script.py
Comment (bbelderbos) [inline uppercase_vowels/script.py:1]: I find GitHub's new commit comment feature quite confusing, the extra feedback is now not visible on this PR, did you get my extra suggestions @xettrisomeman?

File: uppercase_vowels/exercise.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+Dependencies (list any required libraries or leave as "None"):
+GitHub user to credit: xettrisomeman
+---
+
+# Uppercase Vowels
+
+---
+ The Python function `uppercase_vowels` takes in `string` or `filename` and returns `string` with vowels uppercased. The `string` can contain punctuation, uppercase consonants but in the final result vowels should be uppercased and consonants should be lowercased and punctuations should be removed while preserving spaces.
+
+> Extension of `file` should not be uppercased.
```
Comment (bbelderbos) [inline uppercase_vowels/exercise.md]: not sure why this is a markdown quote? Please make this part of the regular text.

File: uppercase_vowels/exercise.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+mp3
+mp4
+pdf
+png
+txt
+exe
+```
+
+
+Example Usage:
```
Comment (bbelderbos) [inline uppercase_vowels/exercise.md:26]: nice addition here

File: uppercase_vowels/exercise.md (hunk 1 above)
Comment (bbelderbos) [inline uppercase_vowels/exercise.md]: I would reword the last part: "punctuations should be removed while preserving spaces" to "each punctuation should be replaced by a single space character (`" "`)"

File: uppercase_vowels/script-template.py [hunk 7]
```diff
@@ -0,0 +1 @@
+def uppercase_vowels(text: str) -> str: ...
```
Comment (bbelderbos) [inline uppercase_vowels/script-template.py:1]: should we include `from string import punctuation` here? I think it's best practice to use this and then we have clarity what punctuation chars to include

**Review by bbelderbos — CHANGES_REQUESTED** [review]

File: uppercase_vowels/script.py
Comment (xettrisomeman) [reply uppercase_vowels/script.py:1]: yes, i got it. I will make the changes. Thanks for the feedback,

File: uppercase_vowels/script.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from string import punctuation
+
+
+def uppercase_vowels(text: str):
+    VOWELS = "aeiou"
+    EXTENSIONS = [".mp3", ".jpg", ".jpeg", ".pdf", ".txt", ".mp4", ".png", ".exe"]
```
Comment (bbelderbos) [inline uppercase_vowels/script.py]: please move those outside of the function

File: uppercase_vowels/script-template.py (hunk 7 above)
Comment (bbelderbos) [reply uppercase_vowels/script-template.py:1]: @xettrisomeman shall we provide this import in the template or would that give away too much?

File: uppercase_vowels/exercise.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
+---
+Level (beginner, intermediate, advanced): beginner
+Tags (2-5): string
```
Comment (bbelderbos) [inline uppercase_vowels/exercise.md]: could add 2 more here

**Review by bbelderbos — CHANGES_REQUESTED** [review]
Thanks, almost there ...

File: uppercase_vowels/script-template.py (hunk 7 above)
Comment (xettrisomeman) [reply uppercase_vowels/script-template.py:1]: no, I don't think it is necessary. Most will find a way to do(manually or using functions)

Comment (bbelderbos) [reply uppercase_vowels/script-template.py:1]: agreed, thanks

**Review by bbelderbos — APPROVED** [review]
Thanks, cool Bite, I will add it to the platform coming week.

**Comment by bbelderbos** [comment]
@xettrisomeman just because it's the first merged PR, can you please confirm you got the score on your hacktober dashboard? Thanks.

**Comment by bbelderbos** [comment]
[Added it to the platform](https://pybitesplatform.com/bites/uppercase-vowels/), nice, thanks!

I did end up including the constants in the template code:

```
VOWELS = "aeiou"
EXTENSIONS = [".mp3", ".jpg", ".jpeg", ".pdf", ".txt", ".mp4", ".png", ".exe"]


def uppercase_vowels(text: str) -> str:
    pass
```
